sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/sns/SnsIO.java

```diff
@@ -62,310 +58,272 @@
  * PCollection<String> data = ...;
  *
  * data.apply(SnsIO.<String>write()
- *     .withPublishRequestFn(m -> PublishRequest.builder().topicArn("topicArn").message(m).build())
- *     .withTopicArn("topicArn")
- *     .withRetryConfiguration(
- *        SnsIO.RetryConfiguration.create(
- *          4, org.joda.time.Duration.standardSeconds(10)))
- *     .withSnsClientProvider(new BasicSnsClientProvider(awsCredentialsProvider, region));
+ * 		.withElementCoder(StringUtf8Coder.of())
+ * 		.withPublishRequestFn(createPublishRequestFn())
```

Comment:
It would be useful to show a simple example of `createPublishRequestFn()` implementation, like you did for tests.